cmd/kubeadm/app/cmd/phases/upgrade/node/kubeletconfig.go

```diff
@@ -0,0 +1,126 @@
+		ConfigMap to determine what the _desired_ kubelet version is, but the user can override this by using the
+		--kubelet-version parameter.
+		`)
+)
+
+// NewKubeletConfigPhase creates a kubeadm workflow phase that implements handling of kubelet-config upgrade.
+func NewKubeletConfigPhase() workflow.Phase {
+	phase := workflow.Phase{
+		Name:  "kubelet-config",
+		Short: "Upgrade the kubelet configuration for this node",
+		Long:  kubeletConfigLongDesc,
+		Run:   runKubeletConfigPhase(),
+		InheritFlags: []string{
+			options.DryRun,
+			options.KubeconfigPath,
+			options.KubeletVersion,
+		},
+	}
+	return phase
+}
+
+func runKubeletConfigPhase() func(c workflow.RunData) error {
+	return func(c workflow.RunData) error {
+		data, ok := c.(Data)
+		if !ok {
+			return errors.New("kubelet-config phase invoked with an invalid data struct")
+		}
+
+		// otherwise, retrieve all the info required for kubelet config upgrade
+		cfg := data.Cfg()
+		client := data.Client()
+		dryRun := data.DryRun()
+
+		// Set up the kubelet directory to use. If dry-running, this will return a fake directory
+		kubeletDir, err := upgrade.GetKubeletDir(dryRun)
+		if err != nil {
+			return err
+		}
+
+		// Gets the target kubelet version.
+		// by default kubelet version is expected to be equal to ClusterConfiguration.KubernetesVersion, but
+		// users can specify a different kubelet version (this is a legacy of the original implementation
+		// of `kubeam upgrade node config` which we are preserving in order to don't break GA contract)
+		kubeletVersionStr := cfg.ClusterConfiguration.KubernetesVersion
+		if data.KubeletVersion() != "" && data.KubeletVersion() != kubeletVersionStr {
+			kubeletVersionStr = data.KubeletVersion()
+			fmt.Printf("[upgrade] Using kubelet config version %s, while kubernetes-version is %s\n", kubeletVersionStr, cfg.ClusterConfiguration.KubernetesVersion)
+		}
+
+		// Parse the desired kubelet version
+		kubeletVersion, err := version.ParseSemantic(kubeletVersionStr)
+		if err != nil {
+			return err
+		}
+
+		// TODO: Checkpoint the current configuration first so that if something goes wrong it can be recovered
+		if err := kubeletphase.DownloadConfig(client, kubeletVersion, kubeletDir); err != nil {
+			return err
+		}
+
+		// If we're dry-running, print the generated manifests
+		if dryRun {
+			if err := printFilesIfDryRunning(dryRun, kubeletDir); err != nil {
+				return errors.Wrap(err, "error printing files on dryrun")
+			}
+			return nil
+		}
+
+		fmt.Println("[upgrade] The configuration for this node was successfully updated!")
+		fmt.Println("[upgrade] Now you should go ahead and upgrade the kubelet package using your package manager.")
+		return nil
+	}
+}
+
+// printFilesIfDryRunning prints the Static Pod manifests to stdout and informs about the temporary directory to go and lookup
+func printFilesIfDryRunning(dryRun bool, kubeletDir string) error {
+	if !dryRun {
```

Comment:
Due  we are calling it  from `if dryRun {}` we can remove this if and rename to `printFiles`